driver/src/price_finding/naive_solver.rs

```diff
@@ -101,8 +102,20 @@ impl PriceFinding for NaiveSolver {
         orders: &[Order],
         state: &AccountState,
     ) -> Result<Solution, PriceFindingError> {
+        // Fetch max_token_id to determine num_tokens
+        let max_token_id = orders
+            .iter()
+            .map(|o| cmp::max(o.buy_token, o.sell_token))
+            .max()
+            .unwrap();
+        let num_tokens = if let Some(fee) = &self.fee {
+            cmp::max(fee.token, max_token_id) + 1
+        } else {
+            max_token_id + 1
+        };
+
         // Initialize trivial solution (default of zero indicates untouched token).
-        let mut prices: Vec<u128> = vec![0; TOKENS as usize];
+        let mut prices: Vec<u128> = vec![0; num_tokens as usize];
```

Comment:
Should this be a Map instead of a vector? Maybe this will imply changes to the Solution struct, which could be done in a separate PR on top?

Reply:
I would very much agree with this. Should the separate PR come first? The chicken or the egg? I would think the change to a map should come first.